src/charm.py

```diff
@@ -165,6 +187,50 @@ def _create_kubeconfigs(self, event):
             token=credentials.get("proxy_token"),
         )
 
+    def _get_metrics_endpoints(self) -> list:
+        """Return the metrics endpoints for K8s components."""
+        log.info("Building Prometheus scraping jobs.")
+
+        cos_user = f"system:cos:{kubernetes_snaps.get_node_name()}"
+        token = self.tokens.get_token(cos_user)
+
+        if not token:
+            log.info("Token not provided by the relation")
+            return []
+
+        def create_scrape_job(config: JobConfig):
+            return {
+                "tls_config": {"insecure_skip_verify": True},
+                "authorization": {"credentials": token},
+                "job_name": config.name,
+                "metrics_path": config.metrics_path,
+                "scheme": config.scheme,
+                "static_configs": [
+                    {
+                        "targets": [config.target],
+                        "labels": {"node": kubernetes_snaps.get_node_name()},
```

Comment:
the node-name may different on aws and should be called with the fqdn

if you rebase you'll spot other calls to `get_node_name` here